### PR TITLE
[SPARK-42684][SQL] v2 catalog should not allow column default value by default

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Following generated columns, column default value should also have a catalog capability and v2 catalogs must explicitly declare SUPPORT_COLUMN_DEFAULT_VALUE to support it.

### Why are the changes needed?

column default value needs dedicated handling and if a catalog simply ignores it, then query result can be wrong.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

new tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gengliangwang @dtenedor

**Review by gengliangwang — APPROVED** [review]
+1, LGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala [hunk 1]
```diff
@@ -137,6 +113,49 @@ object ResolveDefaultColumns {
...
   }
 
+  // Fails if the given catalog does not support column default value.
+  def validateCatalogForDefaultValue(
+      schema: StructType,
+      catalog: TableCatalog,
+      ident: Identifier): Unit = {
+    if (SQLConf.get.enableDefaultColumns &&
+      schema.exists(_.metadata.contains(CURRENT_DEFAULT_COLUMN_METADATA_KEY))) {
+      if (!catalog.capabilities().contains(
```
Comment (dtenedor) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala:123]: We can combine with the previous if statement using `&&`?

**Review by dtenedor — APPROVED** [review]
This is a good safeguard for correctness.

**Comment by cloud-fan** [comment]
thanks for review, merging to master/3.4!

**Pull request closed without merging**